libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1953,12 +1963,18 @@ void AP_InertialSensor::update(void)
         for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
             if (_gyro_healthy[i] && _use(i)) {
                 _first_usable_gyro = i;
+#if !AP_AHRS_ENABLED
+                _primary_gyro = _first_usable_gyro;
+#endif
```

Comment:
It might be a little clearer to drop this and the accel equivalent outside these two loops, but right below them.